src/server/game/Entities/Player/Player.cpp

```diff
@@ -27594,4 +27594,124 @@ bool Player::IsPetDismissed()
     return false;
 }
 
+/* Premium character interaction.
+ * These functions will allow modules to access the new generic
+ * premium tables designed to help modules that interact with
+ * players and characters without the need to create other tables.
+ */
+int Player::GetCharacterPremiumLevel(uint32 guid)
+{
+    PreparedStatement* stmt = CharacterDatabase.GetPreparedStatement(CHAR_SEL_PREMIUM_CHARACTER_PREMIUM_LEVEL);
+    stmt->setUInt32(0, GUID_LOPART(guid));
+
+    PreparedQueryResult result = CharacterDatabase.Query(stmt);
+
+    if (!result)
+        return 0;
+
+    int premium_level = (*result)[0].GetInt8();
+    return premium_level;
+}
+
+
+bool Player::CreateCharacterPremiumLevel(uint32 guid, int premiumLevel)
+{
+    // Validate if character to be inserted already has a premium level
+    int hasPremiumLevel = GetCharacterPremiumLevel(guid);
+
+    if (!hasPremiumLevel)
+    {
+        PreparedStatement* stmt = CharacterDatabase.GetPreparedStatement(CHAR_INS_PREMIUM_CHARACTER);
+        stmt->setUInt32(0, GUID_LOPART(guid));
+        stmt->setInt8(1, premiumLevel);
+
+        PreparedQueryResult result = CharacterDatabase.Query(stmt);
+
+        return true;
+    }
+    else
+        return false;
+}
+
+bool Player::DeleteCharacterPremiumLevel(uint32 guid)
+{
+    // Validate if character to be removed has a premium level
+    int hasPremiumLevel = GetCharacterPremiumLevel(guid);
+
+    if (hasPremiumLevel)
+    {
+        PreparedStatement* stmt = CharacterDatabase.GetPreparedStatement(CHAR_DEL_PREMIUM_CHARACTER);
+        stmt->setUInt32(0, GUID_LOPART(guid));
+
+        PreparedQueryResult result = CharacterDatabase.Query(stmt);
+
+        return true;
+    }
+    else
+        return false;
+}
+
+int Player::GetAccountPremiumLevel(uint32 guid)
+{
+    uint32 accountID = GetCharacterAccount(GUID_LOPART(guid));
```

Comment:
Read above: With an unordered map, this function would be 3 lines and have no database calls.